static/js/click_handlers.js

```diff
@@ -532,6 +535,39 @@ export function initialize() {
         }
     });
 
+    $("body").on("click", "#all_private_messages_icon", (e) => {
+        e.preventDefault();
+        pm_list.activate_all_private_messages_view();
+        top_left_corner.deselect_top_left_corner_items();
+        window.location.href = "#narrow/is/private";
+    });
+
+    $("#toggle_private_messages_section").on("click", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        $("#toggle_private_messages_section_icon").toggleClass(
+            "'fa fa-caret-down fa-lg' 'fa fa-caret-right fa-lg'",
+        );
+        const private_messages_collapsed = pm_list.return_private_messages_state();
+        if (
+            private_messages_collapsed &&
+            ($(".expanded-private-messages").children().length === 0 ||
+                pm_list.only_active_pm_out_of_collapsed_pms)
+        ) {
+            // To expand the PM section list we need the PM section state to be collapsed
+            // then there comes 2 possible conditions regarding the number of PMs present in the section
+            // either it should 0 PMs present or else only 1 which is active PM can be present outside the collapsed PM section.
+            pm_list.expand();
+        } else {
+            // If PM section state is expanded we trigger function to close it and mark the state as collapsed.
+            pm_list.close();
+        }
+        setTimeout(() => {
+            // We resize the STREAMS section after changing the state of PM section.
+            resize.resize_stream_filters_container();
+        }, 0);
+    });
```

Comment:
This is a more complex click handler than we usually aim to have in this file. This click handler should call a function with a name like `pm_list.toggle_private_messages_section`.

Reply:
Sure, would update this in a preparatory commit.